### PR TITLE
Make maximum document size configurable

### LINKED ISSUE #4233: Make maximum documents size configurable
### What should be done?

It appears that FerretDB inherit the 16MiB document size limits from MongoDB. It is not pretty reasonable nowadays.

I think we can have an option to tweak the default limit.

### FerretDB version
v1.21.0

### Backend
PostgreSQL 12.7

### Steps to reproduce
`mongoimport` a document larger than 16MiB to FerretDB. Then `mongoimport` exits with the error `Failed: an inserted document is too large`.

### PR DESCRIPTION
**Pull request by chilagrow**
# Description

Closes #4233.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [x] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handler/msg_hello.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ func (h *Handler) MsgHello(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, e
 	var reply wire.OpMsg
 	resp := must.NotFail(types.NewDocument(
 		"isWritablePrimary", true,
-		"maxBsonObjectSize", int32(types.MaxDocumentLen),
+		"maxBsonObjectSize", int32(h.MaxBsonObjectSizeBytes),
 		"maxMessageSizeBytes", int32(wire.MaxMsgLen),
```
Comment (chilagrow) [inline internal/handler/msg_hello.go:52]: Not in scope of this issue, there is `wire.MaxMsgLen` which is 48MB. So currently entire message size is limited by this https://github.com/chilagrow/FerretDB/blob/4db0c5e5ef79edd7fbe654b173a0e804011a7465/internal/wire/msg_header.go#L99

I'm noting that if this check on wire level is something we might want to change in the future?

File: integration/insert_command_test.go [hunk 2]
```diff
@@ -195,3 +196,23 @@ func TestInsertIDDifferentTypes(t *testing.T) {
 		err,
 	)
 }
+
+func TestInsertTooLargeDocument(t *testing.T) {
+	if setup.IsMongoDB(t) {
+		t.Skip("Skipping for MongoDB: using setup.FailsForMongoDB takes too long, 120s to fail by server selection timeout error.")
+	}
```
Comment (chilagrow) [inline integration/insert_command_test.go:203]: Suggestion welcome, if we want to handle this otherwise

**Review by noisersup — APPROVED** [review]

**Review by rumyantseva — APPROVED** [review]
Looks good!

**Comment by vsisl** [comment]
Hi @chilagrow 

Could you point me to instructions how to set the maximum document size when running ferredDB in docker using docker?

Thank you for your help! :)

**Comment by vsisl** [comment]
Update:

I found this [blog post](https://blog.ferretdb.io/new-ferretdb-v122-user-setup-feature/#make-maximum-document-size-configurable) and modified my docker compose to:


```yml
services:
  postgres:
    image: postgres
    restart: on-failure
    environment:
      - POSTGRES_USER=username
      - POSTGRES_PASSWORD=password
      - POSTGRES_DB=ferretdb
    volumes:
      - ./data:/var/lib/postgresql/data

  ferretdb:
    image: ghcr.io/ferretdb/ferretdb
    restart: on-failure
    ports:
      - 127.0.0.1:27017:27017
    environment:
      - FERRETDB_POSTGRESQL_URL=postgres://postgres:5432/ferretdb
      - FERRETDB_TEST_BATCH_SIZE=537395136 # 512 MB
      - FERRETDB_TEST_MAX_BSON_OBJECT_SIZE_MI_B=537395136 # 512 MB

networks:
  default:
    name: ferretdb
```

However, when trying to insert a document using pymongo, I get the following error:

```
  File "/Users/.../lib/python3.11/site-packages/pymongo/message.py", line 1281, in _raise_document_too_large
    raise DocumentTooLarge(f"{operation!r} command document too large")
pymongo.errors.DocumentTooLarge: 'saslStart' command document too large
```
 
 Any ideas?

**Comment by chilagrow** [comment]
@vsisl you might have accidentally configured it too large. Try something more suitable like below, the unit is MiB :hugs: 

```
      - FERRETDB_TEST_MAX_BSON_OBJECT_SIZE_MI_B=512 # 512 MiB
```

As for `FERRETDB_TEST_BATCH_SIZE`, let's use the default by removing the option. It's the number of documents to be inserted at once by backend, not related to the size. So `5373951361` would be too many.

**Comment by vsisl** [comment]
Hi @chilagrow 

Thank you very much for your quick reply! I really appreciate your help! :)

I followed your instructions but ran into another problem. I'm now getting a connection timeout error. This is strange because the error message mentions 20s timeout. But in reality, the error is raised immediately after executing the `.insert_one()` method. There's definitely now 20s gap before the error is raised.

I imagine that this is probably not the right place to seek help with this. Nevertheless, I decided to post this here, just in case you've seen such an error before...

Thanks again for your help! 

```
  File "/Users/.../lib/python3.11/site-packages/pymongo/collection.py", line 658, in insert_one
    self._insert_one(
  File "/Users/.../lib/python3.11/site-packages/pymongo/collection.py", line 598, in _insert_one
    self.__database.client._retryable_write(
  File "/Users/.../lib/python3.11/site-packages/pymongo/mongo_client.py", line 1575, in _retryable_write
    return self._retry_with_session(retryable, func, s, bulk, operation, operation_id)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/.../lib/python3.11/site-packages/pymongo/mongo_client.py", line 1461, in _retry_with_session
    return self._retry_internal(
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/.../lib/python3.11/site-packages/pymongo/_csot.py", line 108, in csot_wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/.../lib/python3.11/site-packages/pymongo/mongo_client.py", line 1507, in _retry_internal
    ).run()
      ^^^^^
  File "/Users/.../lib/python3.11/site-packages/pymongo/mongo_client.py", line 2353, in run
    return self._read() if self._is_read else self._write()
                                              ^^^^^^^^^^^^^
  File "/Users/.../lib/python3.11/site-packages/pymongo/mongo_client.py", line 2470, in _write
    return self._func(self._session, conn, self._retryable)  # type: ignore
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/.../lib/python3.11/site-packages/pymongo/collection.py", line 586, in _insert_command
    result = conn.command(
             ^^^^^^^^^^^^^
  File "/Users/.../lib/python3.11/site-packages/pymongo/helpers.py", line 342, in inner
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/.../lib/python3.11/site-packages/pymongo/pool.py", line 1017, in command
    self._raise_connection_failure(error)
  File "/Users/.../lib/python3.11/site-packages/pymongo/pool.py", line 1214, in _raise_connection_failure
    _raise_connection_failure(self.address, error, timeout_details=details)
  File "/Users/.../lib/python3.11/site-packages/pymongo/pool.py", line 427, in _raise_connection_failure
    raise AutoReconnect(msg) from error
pymongo.errors.AutoReconnect: localhost:27017: [Errno 32] Broken pipe (configured timeouts: connectTimeoutMS: 20000.0ms)
```

**Comment by chilagrow** [comment]
@vsisl thanks for letting us know. We would like a bit more information, could you file a [bug](https://github.com/FerretDB/FerretDB/issues/new?assignees=ferretdb-bot&labels=code%2Fbug%2Cnot+ready&projects=&template=bug.yml) please :hugs:. Don't forget to include the version of pymongo driver and logs from FerretDB too, those are helpful for us!

**Pull request merged**